backend/src/services/user.service.js

```
@@ -35,7 +52,37 @@ class UserService {
         throw new Error("User ID and updates are required");
       }
-      const user = await UserModel.updateUser(id, updates);
+
+      const numericId = parseInt(id, 10);
+
+      if (numericId !== updates.id) {
+        throw new Error("User ID in updates does not match the provided ID");
+      }
+
+      // delete id from updates to avoid conflicts
+      delete updates.id;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)